packages/integration_test/lib/src/reporter.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+// ignore: implementation_imports
+import 'package:test_api/src/backend/live_test.dart';
+// ignore: implementation_imports
+import 'package:test_core/src/runner/engine.dart';
+// ignore: implementation_imports
+import 'package:test_core/src/runner/reporter.dart';
+
+import '../common.dart';
+import 'constants.dart';
+
+/// A reporter that plugs into [directRunTests] from `package:test_core`.
+class ResultReporter implements Reporter {
+  /// When the [_engine] has completed execution of tests, [_resultsCompleter]
+  /// will be completed with the test results.
+  ResultReporter(this._engine, this._resultsCompleter) {
+    _subscriptions.add(_engine.success.asStream().listen(_onDone));
+  }
+  final Engine _engine;
+  final Completer<Map<String, Object>> _resultsCompleter;
+
+  final Set<StreamSubscription<Object>> _subscriptions = <StreamSubscription<Object>>{};
+
+  void _onDone(bool _) {
+    _cancel();
+    final Map<String, Object> results = <String, Object>{
+      for (final LiveTest liveTest in _engine.liveTests)
+        liveTest.test.name: liveTest.state.result.name == success
+            ? success
+            : Failure(
+                liveTest.test.name, liveTest.errors.first.stackTrace.toString(),
+                error: liveTest.errors.first.error)
```

Comment:
```suggestion
            : Failure(
                liveTest.test.name, 
                liveTest.errors.first.stackTrace.toString(),
                error: liveTest.errors.first.error,
              )
```

Also, this seems concerning that it will only have the first error.  It seems like it would be better to capture them all.

Reply:
Resolved the formatting issue.

> Also, this seems concerning that it will only have the first error. It seems like it would be better to capture them all.

I took this snippet from the Google3 test reporter. Will do a bit of research to find out if this is necessary and follow up :)

Reply:
@natebosch may know.

Reply:
I think in any system which expects only a single reason for a test case to fail it is sensible to choose the first error to report. The Dart test runner allows a single test to have multiple errors, but if we are reporting into a system that only allows a single reason to fail we have pick between trying to do a strange concatenation or only reporting the first error.

Reply:
I think we should probably support whatever errors we can access internally, and then we can either bundle them up or not when we report to JUint/XCTest/etc.

Reply:
Made this collect the list of errors. I did not wire the errors into the platform side as it's a bit out of scope.